src/widgets/dock_area/mod.rs

```diff
@@ -1063,8 +1063,8 @@ impl<'tree, Tab> DockArea<'tree, Tab> {
                 tabs_style.tab_body.stroke,
             );
 
-            if self.scroll_area_in_tabs {
-                ScrollArea::both().show(ui, |ui| {
+            if self.scroll_area_in_tabs && tab_viewer.tab_scroll(tab) {
```

Comment:
Once you delete `tab_scroll`, you can delete this `if`, the whole `else` branch, and also the `scoll_area_in_tabs` altogether. The presence of scroll bars will then only be controlled by `TabViewer::scroll_both`.